definitions/tools/extract_hla_alleles.wdl

```diff
@@ -11,10 +12,17 @@ task extractHlaAlleles {
     docker: "ubuntu:xenial"
     disks: "local-disk ~{space_needed_gb} HDD"
   }
-
+  
+  # first, extract HLA class I from the optitype file
+  # second, extract HLA class II from the phlat file
+  # third, ensure there are only 2 fields of accuracy for alleles
+ 
   String outname = "helper.txt"
+  String temp = "temp.txt"
   command <<<
-    /usr/bin/awk '{FS="\t";getline;for(n=2;n<=NF-2;n++){if($n==""){}else{printf "HLA-"$n"\n"}}}' ~{file} > ~{outname}
+    /usr/bin/awk '{FS="\t";getline;for(n=2;n<=NF-2;n++){if($n==""){}else{printf "HLA-"$n"\n"}}}' ~{file} > ~{temp}
+    grep "HLA_D" ~{phlat_file} | /usr/bin/awk '{FS="\t";if($2==""){}else{printf $2"\n"};if($3==""){}else{printf $3"\n"}}' >> ~{temp}
+    /usr/bin/awk -F":" '{print $1 ":" $2}' ~{temp} > ~{outname}
```

Comment:
If you want, this can be done without a temporary file by grouping the commands together for the pipe.  Something like:

```bash
(/usr/bin/awk ... ~{file}; grep ... ~{phlat_file} | /usr/bin/awk ...) | /usr/bin/awk ...
```